Source/DafnyLanguageServer/Language/DiagnosticErrorReporter.cs

```diff
@@ -14,8 +15,20 @@ public class DiagnosticErrorReporter : ErrorReporter {
     private readonly DocumentUri entryDocumentUri;
     private readonly Dictionary<DocumentUri, List<Diagnostic>> diagnostics = new();
```

Comment:
Have you considered using a thread-safe dictionary? https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-6.0

Reply:
I thought it wouldn't help, since there's concurrency on the access to the dictionary values as well (they are mutable lists), and since we have two separate dictionaries to update at the same time.